### ISSUE TITLE
Set up database and create form that submits company ratings to database

### ISSUE THREAD

**Issue by cattsmall** [GoodForPoC/website#3]
Requires all questions from this form: https://docs.google.com/forms/d/e/1FAIpQLScCZttwiwtycB7ifLZpXBCZryo2hHFKcbCmYn8EHX7ohuYq3g/viewform?c=0&w=1

**Comment by 0xStarcat** [comment]
Hi! This seems like a good use case for a Rails API with a PostgresDB. What do people think?

**Comment by 0xStarcat** [comment]
We had a discussion on Gitter with @cattsmall and @jalcine and decided on a few basic architecture decisions: 

### Hosting: 

- Ubuntu 17.01 Server on Digital Ocean

### Database:
- PSQL

### Backend server/API:

- Sinatra w/ GraphQL

### Front-End:

- EmberJS

**Comment by 0xStarcat** [comment]
some general whiteboarding being done here - https://sketchboard.me/VAFHP1lgrOJW#/ 

ask me or one of the team members for editing access =)

**Comment by cattsmall** [comment]
Jacky created an architecture wiki post: https://github.com/GoodForPoC/GoodForPoC/wiki/Architecture

Feel free to discuss it here!

**Comment by cattsmall** [comment]
Hey all! I set up Sinatra and GraphQL in a Gemfile + did a fresh install of EmberJS. It's been pushed in the newest file. I'll make sure to reference this issue in future posts. 

Feel free to take it from here and get the backend set up beyond that!

**Comment by jalcine** [comment]
To ensure reproducibilty and transparency of the infra, we're going to define some server provisioning scripts. I'll open another issue about this (and end up working on this, lol).

**Comment by jalcine** [comment]
Tight, thanks Catt!

**Comment by jalcine** [comment]
I'm going to open a new repository that'll hold the API that this client would hit. In that repository, it'll have the infrastructure building code there as well. I'll track those things there and make it a bocker for this.

**Comment by jalcine** [comment]
The new repository is here: https://github.com/goodforpoc/goodforpoc-api.

**Comment by jalcine** [comment]
So, I don't wanna block y'all re: getting data. We can use some tooling to make a "mock" resource for y'all (using mirage + pretender).

[1]: http://www.ember-cli-mirage.com/
[2]: https://github.com/rwjblue/ember-cli-pretender

Once the API's fully ready; it'd be a hot swap - won't notice a thing :)

**Comment by jalcine** [comment]
Got some code here that'll set up a mock GraphQL endpoint. https://github.com/GoodForPoC/GoodForPoc-Api/pull/10.

**Comment by jalcine** [comment]
So the above is done (re: https://github.com/GoodForPoC/GoodForPoC-API/pull/10).

Now we'd just have to have the frontend use models that communicate with this.

**Comment by jalcine** [comment]
Most of the frontend models needed are in.

**Comment by cattsmall** [comment]
Is this still blocked?

**Comment by jalcine** [comment]
A bit, but we can get around it.

## Direct Unblocking Route
- [ ] Have API provide some tests to show how to post a model.
- [ ] Provide tests showing failures to give examples on how to handle it.

## Concurrent Unblocking Route
Use [`ember-cli-mirage`][1] and [this method in this blog post][2] to work on GraphQL-esque stuff without having to wait on the API to fully support it.

[1]: http://www.ember-cli-mirage.com/
[2]: https://emberway.io/faking-graphql-with-ember-cli-mirage-811e83937a5b

**Comment by cattsmall** [comment]
@jalcine this makes sense to me! Godspeed, and let me know how I/others can help!

**Comment by jalcine** [comment]
Will do. I'll add information about the direct unblocking route since it's better long-term.
